src/components/Onboarding/Onboarding.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect } from 'react';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { Image, SafeAreaView, View } from 'react-native';
+import Onboarding from 'react-native-onboarding-swiper';
+import { useNavigation } from '@react-navigation/native';
+import { useStyles } from 'react-native-unistyles';
+import { useTranslation } from 'react-i18next';
+
+import { styleSheet } from './Onboarding.style';
+import useGetOnboardingStatus from '@src/hooks/useGetOnboardingStatus';
+import { DotIndicator } from './OnboardingButtons/Indicator';
+import { Next } from './OnboardingButtons/Next';
+
+import Onboarding1 from '@assets/images/onboarding/onboarding1.png';
+import Onboarding2 from '@assets/images/onboarding/onboarding2.png';
+import Onboarding3 from '@assets/images/onboarding/onboarding3.png';
+
+export const OnboardingScreen = () => {
+	const { isFirstLaunch, isLoading: onboardingIsLoading } = useGetOnboardingStatus();
+	const { t } = useTranslation();
+	const { styles } = useStyles(styleSheet);
+	const navigation = useNavigation();
+
+	useEffect(() => {
+		if (!onboardingIsLoading && !isFirstLaunch) {
+			(navigation as any).navigate('login/index');
+		}
+	}, [isFirstLaunch, onboardingIsLoading, navigation]);
+
+	const setOnboardingStatus = async () => {
+		try {
+			await AsyncStorage.setItem('hasOnboardingSeen', 'false');
+		} catch (error) {
+			// Error saving data
+			console.log(error);
+		}
+	};
```

Comment:
This code is a bit fishy, since after changing AsyncStorage value you techicaly end up in a position when you are out if sync with state inside `useGetOnboardingStatus` hook

Reply:
I understand that it feels like it is highly unlikely to manifest itself, but you can end up in a very messy app state in the end

Reply:
It is also a good idea to put 'hasOnboardingSeen' in some sort of constant to avoid errors in using storage values